src/trans.c

```diff
@@ -1500,16 +1500,23 @@ Obj FuncON_KERNEL_ANTI_ACTION (Obj self, Obj ker, Obj f, Obj n) {
   UInt    deg, i, j, rank, len;
   Obj     out;
 
-  if (INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {
+  assert(IS_LIST(ker) && IS_INTOBJ(n));
+
+  len = LEN_LIST(ker);
+  if (len == 1 && INT_INTOBJ(ELM_LIST(ker, 1)) == 0) {
     return FuncFLAT_KERNEL_TRANS_INT(self, f, n);
   }
 
-  len = LEN_LIST(ker);
   rank = 1;
 
   if (TNUM_OBJ(f) == T_TRANS2) {
     deg = INT_INTOBJ(FuncDegreeOfTransformation(self, f));
     if (len >= deg) {
+      if (len == 0) {
+        out = NEW_PLIST(T_PLIST_EMPTY + IMMUTABLE, len);
+        SET_LEN_PLIST(out, len);
```

Comment:
You know len is 0 here, but the compiler has to work extra to use that; why not replace len by 0 then?